### ISSUE TITLE
Porting OpenBVE to WebAsssembly

### ISSUE THREAD

**Issue by afarlie** [leezer3/OpenBVE#227]
This is probably impossible, but would there be any interest from the developers in porting a simplified route previewer or object viewer to WebAsm so it can be used in a browser?

**Comment by afarlie** [comment]
http://webassembly.org/

**Comment by Sirflankalot** [comment]
With the current implementation of OpenBVE and friends, I'm afraid it is basically impossible. Now you can go C# to wasm, through the use of mono, but you would have to substitute the renderer with something else for the wasm backend, and that is basically impossible. I've tried to do it in order to move to more modern opengl, and I decided it was an uphill battle that I was going to lose. Additionally we may run into massive performance problems with wasm due to the implementation of C# engines on wasm.

I do have a project that shows a bit of promise, rewriting most of the core of the game in a more modular format in C++. It is a large amount of effort and I have school to attend to so progress isn't going as fast as I would like, but this project would have the possibility of allowing other platforms besides desktop to participate in OpenBVE, as well as help build tools and stuff for interaction with the game and it's files. You can find a link [here](https://github.com/Sirflankalot/openbve2). 

TL;DR: Not now, but maybe "soon".

**Comment by Sirflankalot** [comment]
@leezer3 I think we can close this for now.
